osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -441,138 +458,123 @@ private void updateState(CatcherAnimationState state)
+        private void addLighting(CatchHitObject hitObject, float x, Color4 colour)
         {
-            if (!hitLighting.Value) return;
-
             HitExplosion hitExplosion = hitExplosionPool.Get();
-            hitExplosion.HitObject = caughtObject.HitObject;
-            hitExplosion.X = caughtObject.X;
-            hitExplosion.Scale = new Vector2(caughtObject.HitObject.Scale);
-            hitExplosion.ObjectColour = caughtObject.AccentColour.Value;
+            hitExplosion.HitObject = hitObject;
+            hitExplosion.X = x;
+            hitExplosion.Scale = new Vector2(hitObject.Scale);
+            hitExplosion.ObjectColour = colour;
             hitExplosionContainer.Add(hitExplosion);
         }
 
-        private DrawablePalpableCatchHitObject createCaughtObject(PalpableCatchHitObject source)
+        private CaughtObject getCaughtObject(PalpableCatchHitObject source)
         {
             switch (source)
             {
-                case Banana banana:
-                    return new DrawableBanana(banana);
+                case Fruit _:
+                    return caughtFruitPool.Get();
 
-                case Fruit fruit:
-                    return new DrawableFruit(fruit);
+                case Banana _:
+                    return caughtBananaPool.Get();
 
-                case TinyDroplet tiny:
-                    return new DrawableTinyDroplet(tiny);
-
-                case Droplet droplet:
-                    return new DrawableDroplet(droplet);
+                case Droplet _:
+                    return caughtDropletPool.Get();
 
                 default:
                     return null;
             }
         }
 
-        private void clearPlate(DroppedObjectAnimation animation)
+        private CaughtObject getDroppedObject(CaughtObject caughtObject)
         {
-            var caughtObjects = caughtFruitContainer.Children.ToArray();
-            caughtFruitContainer.Clear(false);
+            var droppedObject = getCaughtObject(caughtObject.HitObject);
 
-            droppedObjectTarget.AddRange(caughtObjects);
+            droppedObject.CopyStateFrom(caughtObject);
+            droppedObject.Anchor = Anchor.TopLeft;
+            droppedObject.Position = caughtFruitContainer.ToSpaceOfOtherDrawable(caughtObject.DrawPosition, droppedObjectTarget);
 
-            foreach (var caughtObject in caughtObjects)
-                drop(caughtObject, animation);
+            return droppedObject;
         }
 
-        private void removeFromPlate(DrawablePalpableCatchHitObject caughtObject, DroppedObjectAnimation animation)
+        private void clearPlate(DroppedObjectAnimation animation)
         {
-            if (!caughtFruitContainer.Remove(caughtObject))
-                throw new InvalidOperationException("Can only drop a caught object on the plate");
+            var caughtObjects = caughtFruitContainer.Children.ToArray();
+            var droppedObjects = caughtObjects.Select(getDroppedObject).ToArray();
```

Comment:
Not sure about this double copy. Is there a reason to not just immediately `.Select(getDroppedObject)`, therefore allocating just one array?